4.preprocess_features/scripts/0.convert_cytotable.py

```diff
@@ -0,0 +1,187 @@
+
+
+# # Load in converted profiles to update
+
+# In[4]:
+
+
+# Directory with converted profiles
+converted_dir = pathlib.Path(f"{output_dir}/converted_profiles")
+
+for file_path in converted_dir.iterdir():
+    # Load the DataFrame from the Parquet file
+    df = pd.read_parquet(file_path)
+
+    # If any, drop rows where "Metadata_ImageNumber" is NaN (artifact of cytotable)
+    df = df.dropna(subset=["Metadata_ImageNumber"])
+    
+    # Rearrange columns and add "Metadata" prefix in one line
+    df = df[
+        [
+            "Nuclei_Location_Center_X",
+            "Nuclei_Location_Center_Y",
+            "Cells_Location_Center_X",
+            "Cells_Location_Center_Y",
+            "Image_Count_Cells",
+        ]
+        + [col for col in df.columns if col not in [
+            "Nuclei_Location_Center_X",
+            "Nuclei_Location_Center_Y",
+            "Cells_Location_Center_X",
+            "Cells_Location_Center_Y",
+            "Image_Count_Cells"]]
+    ].rename(
+        columns=lambda col: "Metadata_" + col if col in [
+            "Nuclei_Location_Center_X",
+            "Nuclei_Location_Center_Y",
+            "Cells_Location_Center_X",
+            "Cells_Location_Center_Y",
+            "Image_Count_Cells"
+        ] else col
+    )
+
+    # Save the processed DataFrame as Parquet in the same path
+    df.to_parquet(file_path, index=False)
+
+
+# ## Check output to confirm process worked
+# 
+# To confirm the number of single cells is correct, please use any database browser software to see if the number of rows in the "Per_Cells" compartment matches the number of rows in the data frame.
+
+# In[5]:
+
+
+converted_df = pd.read_parquet(
+    f"./data/converted_profiles/{plate_names[0]}_sc_converted.parquet"
+)
+
+print(converted_df.shape)
+converted_df.head()
+
+
+# ## Extract organoid only profiles
+
+# In[6]:
+
+
+for file_path in sqlite_dir.iterdir():
+    output_path = pathlib.Path(
+        f"{output_dir}/converted_profiles/{file_path.stem}_organoid_converted.parquet"
+    )
+    print("Starting conversion with cytotable for plate:", file_path.stem)
+
+    # Merge single cells and output as parquet file
+    convert(
+        source_path=str(file_path),
+        dest_path=str(output_path),
+        dest_datatype=dest_datatype,
+        metadata=["image"],
+        compartments=["organoids"],
+        identifying_columns=["ImageNumber"],
+        joins="""
+        SELECT
+            *
+        FROM
+            read_parquet('per_image.parquet') as per_image
+        INNER JOIN read_parquet('per_organoids.parquet') AS per_organoids ON
+            per_organoids.Metadata_ImageNumber = per_image.Metadata_ImageNumber
+        """,
+        page_keys={
+            "image": "ImageNumber",
+            "organoids": "Organoids_Number_Object_Number",
+            "join": "Organoids_Number_Object_Number",
+        },
+        chunk_size=10,
+    )
+
+print("All plates have been converted with cytotable!")
```

Comment:
I do include a print statement in the loop that says which plate is being processed, so technically, it is already doing this in the loop. There is a way to add like an `assert` or something to confirm all plates were processed but since this project only has one plate for now processed, we should be good!